build/rust_toolchains.bzl

```diff
@@ -0,0 +1,41 @@
+load("@rules_rust//crate_universe:repositories.bzl", "crate_universe_dependencies")
+load("@rules_rust//rust:repositories.bzl", "rules_rust_dependencies", "rust_register_toolchains", "rustfmt_toolchain_repository")
+load("@rules_rust//tools/rust_analyzer:deps.bzl", "rust_analyzer_dependencies")
+load("//deps/rust/crates:crates.bzl", "crate_repositories")
+
+RUST_STABLE_VERSION = "1.86.0"  # LLVM 19
+
+RUST_NIGHTLY_VERSION = "nightly/2025-02-20"
+
+RUST_TARGET_TRIPLES = [
+    # Add support for macOS cross-compilation
+    "x86_64-apple-darwin",
+    # Add support for macOS rosetta
+    "aarch64-unknown-linux-gnu",
+]
+
+def rust_toolchains():
+    rules_rust_dependencies()
+    rust_register_toolchains(
+        edition = "2021",
+        extra_target_triples = RUST_TARGET_TRIPLES,
+        versions = [
+            RUST_STABLE_VERSION,
+            RUST_NIGHTLY_VERSION,
+        ],
+    )
+
+    for t in RUST_TARGET_TRIPLES:
+        rustfmt_toolchain_repository(
```

Comment:
Not sure if this will work as intended: Here we set up toolchains for the two extra triples, but not for the local target triple. Would having one toolchain for just the local triple be sufficient here? If I interpret the docs correctly we might just be able to drop the exec_triple parameter then.

Reply:
hm. let me try it.

Reply:
`Error: rustfmt_toolchain_repository() missing 1 required positional argument: exec_triple`

Reply:
frankly I am not sure how it picks up nightly rustfmt for me locally then since I don't specify x64 triple?

Reply:
Oof. So there's no way to set it up for just the current target? Maybe it's fine though, CI passed on this PR so maybe if we don't set up a toolchain for the host target rustfmt still works?